### DISCUSSION TITLE
Spline UI "Server is not available" in the execution events and Data Sources section

### DISCUSSION THREAD

**Discussion by VishweshwarKammari** [AbsaOSS/spline#981] [Q&A] [answered]
I have deployed spline on master node of EMR for a POC. I have followed the process as per the documentation for docker deployment. I don't see any errors in the logs. Able to see spline rest server, spline ui on the respective ports. I have started a pyspark shell using the spark agent jar as below. I can not see any lineage generated in ui. It is displaying server error "Server is unavailable". I am new to spline. Kindly help me fixing this issue.

I am using spline 0.6.0 version with the following process.

```
wget https://raw.githubusercontent.com/AbsaOSS/spline-getting-started/main/docker/docker-compose.yml

wget https://raw.githubusercontent.com/AbsaOSS/spline-getting-started/main/docker/.env

docker-compose up

```

![image](https://user-images.githubusercontent.com/55391272/137718096-33a5702f-baef-4b63-b617-6fa08bc89a28.png)

` pyspark   --jars spark-2.4-spline-agent-bundle_2.11-0.6.4.jar --conf "spark.sql.queryExecutionListeners=za.co.absa.spline.harvester.listener.SplineQueryExecutionListener"   --conf "spark.spline.producer.url=http://localhost:8080/producer" --conf "spark.sql.queryExecutionListeners=za.co.absa.spline.harvester.listener.SplineQueryExecutionListener"`

**Comment by wajda** [comment]
Open a development console in your browser (F12) and look at the Network tab. You probably have the server IP inaccessible for your browser.

**Reply by varankarpranay** [reply]
Hi Wajda, I also trying to deploy spline on Kubernetes and Docker VM  but getting 404 error for "/consumer" & "/producer" link in browser. attaching docker steps for better understanding.

docker run -d -p 8529:8529 -e ARANGO_NO_AUTH=1 arangodb/arangodb:3.8.1
docker run -it --rm absaoss/spline-admin:0.7.4 db-init arangodb://172.17.0.1:8529/spline
docker container run  -d -e SPLINE_DATABASE_CONNECTION_URL=arangodb://172.17.0.1:8529/spline -p 8080:8080 absaoss/spline-rest-server:0.7.4

i have login inside spline docker container and try to hit the url, it give me same 404 error. please refer attached screenshot.

![spline-deployment-error](https://user-images.githubusercontent.com/52127002/137762969-551bea0c-647a-4386-9b69-b847f2e0293f.jpeg)

**Comment by wajda** [comment]
It's correct. There is nothing on the `http://172.17.0.1:8080/consumer` URL - it's just a base URL for the other _Consumer REST API_ resources.
To check the server status try `curl http://172.17.0.1:8080/about/version` or `curl -I http://localhost:8080/about/liveness`
If you still get 404 check Catalina logs in `/usr/local/tomcat/logs` inside the Spline server container.

**Reply by varankarpranay** [reply]
i was able to get a response when i hit "/about/version" on spline-rest-server but when i deploy UI getting "server is unavailable" on web browser. i was trying to access UI from internal network. attaching a error for your reference.

![image](https://user-images.githubusercontent.com/52127002/137902106-b5e7da74-f108-4415-907a-943b6e21aef4.png)

**Reply by wajda** [reply]
Is there any details in the browser developer tool?
I don't know what browser are you using, but usually it's opened by pressing `F12`.

Look for the Networking tab and failed AJAX requests. Alternatively check the Console tab.

Example of the Google Chrome:

![image](https://user-images.githubusercontent.com/795479/137909874-ee84411b-18d6-4376-8d6e-54b561e16cf2.png)

**Reply by varankarpranay** [reply]
I am using chrome browser. look like i am getting "Cross-Origin Request Blocked" error in console tab and also getting 304 error for some JS in networking tab.

Any idea how to resolve this CORS error, currently my spline deployed on kubernetes cluster on EKS.

![image](https://user-images.githubusercontent.com/52127002/137934952-0282fddc-55cd-46aa-8da4-27d3830dc897.png)

![image](https://user-images.githubusercontent.com/52127002/137935076-dc2aad22-62cb-4cfa-877e-83372d34c6d0.png)

**Comment by VishweshwarKammari** [comment]
Hi @wajda,

Thanks for your inputs and I am able to access UI and see sample lineage. But When I launched a pyspark job with below configuration and tried to run a small job to load sample data to S3. Lineage was not delivered to spline UI and got http 500 error code. I have attached the screenshot below. Please let me know what could be the cause of this issue.

```
pyspark   --jars spark-3.1-spline-agent-bundle_2.12-0.6.4.jar \
--conf "spark.sql.queryExecutionListeners=za.co.absa.spline.harvester.listener.SplineQueryExecutionListener" \
--conf "spark.spline.producer.url=http://internal-k8s-default-splinein-4abb335d06-1548973854.us-west-2.elb.amazonaws.com:8081/producer" 
```
![image](https://user-images.githubusercontent.com/55391272/137949477-3f2bd92c-c8bf-4374-b742-03e5fdf22dd7.png)

**Comment by wajda** [comment, marked as answer]
Interesting, we didn't have issues with CORS before. How did you fix it in the end?

Regarding the server error, look at the server logs (`/usr/local/tomcat/logs/` inside the container) for the same UUID that you receive on the client (`"errorId"` property).

**Comment by VishweshwarKammari** [comment]
We are using deploying spline on EKS and earlier in our configuration we were using private ips we replaced it with the load balancer and the issue with the communication was resolved.

We looked into the logs of tomcat in the EKS cluster where spline is currently deployed. We have searched in the logs using the error UUID as you suggested.  We are using Arangodb 3.8.1 which is latest. Below is the error that we are seeing. Does this sound familiar to you. Appreciate any suggestions.

![image](https://user-images.githubusercontent.com/55391272/137966472-e1bf448e-a018-4509-b0ef-0594b69a6961.png)

**Comment by wajda** [comment]
Oops, looks like you've spotted an oversight. Thank you! :+1:
It's a missing collection declaration in the AQL `WITH` statement.
I've fixed it in the #984
Could you please test the PR and let me know if it works?

(To build the project with the Docker image use `mvn install -Ddocker -Ddockerfile.repositoryUrl=my`)

**Reply by wajda** [reply]
The reason why we didn't noticed it is because this statement is optional in a non-cluster ArangoDB setup that is used on our CI and development environments.

**Reply by varankarpranay** [reply]
Is Spline deployment support cluster ArangoDB setup? also is the reason we get "AQL: collection not known to traversal: 'dataSources'" error because we used arangodb cluster mode in EKS?

FYI,
apiVersion: "database.arangodb.com/v1"
kind: "ArangoDeployment"
metadata:
  name: "arangodb-cluster"
spec:
  mode: Cluster
  image: 'arangodb/arangodb:3.8.1'
  environment: Production
  externalAccess:
    type: NodePort
  tls:
    caSecretName: None

**Comment by wajda** [comment]
Yes, that particular error manifests itself on a cluster ArangoDB setup only.
Does #984 fixes the issue for you?

**Comment by VishweshwarKammari** [comment]
Hi @Wajda,

We are using spline 0.6.0 and following docker deployment process. We are not using maven build as we are deploying this in EKS. I am not sure where this update/fix has to be made for the deployment method that we are following. Below is the method we followed for deploying spline 0.6.0. We have used docker-compose.yml file and converted that to a kubernetes yaml file for deploying it in EKS.

Please suggest.

```
wget https://raw.githubusercontent.com/AbsaOSS/spline-getting-started/main/docker/docker-compose.yml

wget https://raw.githubusercontent.com/AbsaOSS/spline-getting-started/main/docker/.env

```

**Reply by wajda** [reply]
Don't worry, I've tested it, the issue is fixed. An updated version (0.7.5) will be available in the public repos tomorrow.

**Reply by VishweshwarKammari** [reply]
That's  a great news. We will start implementing it as soon as it is deployed. We will be waiting to hear from you on this!

**Reply by wajda** [reply]
Version 0.7.5 is out - https://hub.docker.com/layers/absaoss/spline-rest-server/0.7.5/images/sha256-36386fe1cf93c6b1a9cea613772496f94ed4f2201cb4bc7a0ad163915f09373b?context=explore

**Reply by VishweshwarKammari** [reply]
Hi @wajda,

Thank you so much for helping us out on this issue. We had deployed spline rest server 0.7.5 and it is working as expected. Thank you for responding quickly to our issues and fixing the bugs.
